docs/import/upload_image_collection.js

```diff
@@ -178,35 +164,41 @@ function uploadFileToGCS(name, contents, collectionName, callback) {
     headers: gcsHeader,
     body: contents,
   };
-  fetch(URL, request)
-      .then((r) => r.json())
-      .then((r) => {
-        uploadedToGCS++;
-        return r;
-      })
-      .catch((err) => {
-        console.error(err);
-        resultDiv.innerHTML += '<br>Error uploading ' + name + ': ' + err;
-        throw err;
-      })
-      .then((result) => callback(result.bucket, collectionName, name))
-      .catch((err) => {
-        console.error(err);
-        resultDiv.innerHTML += '<br>Error processing ' + name + ': ' + err;
-      });
+  let bucket;
+  try {
+    const result = await fetch(URL, request);
+    const json = await result.json();
+    if (result.status !== 200) {
+      throw new Error('Unable to fetch from GCS: ' + json.error.message);
+    }
+    ({bucket} = json);
```

Comment:
what are these outer parens for ?

Reply:
Necessary for destructuring. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment "Assignment without Declaration" section.